usr/src/uts/common/fs/smbsrv/smb2_qinfo_fs.c

```diff
@@ -287,3 +297,113 @@ smb2_qfs_obj_id(smb_request_t *sr)
 {
 	return (NT_STATUS_INVALID_PARAMETER);
 }
+
+/*
+ * Not sure yet where these should go.
+ * Flags in FileFsSectorSizeInformation
+ */
+
+#define	SSINFO_FLAGS_ALIGNED_DEVICE	0x00000001
+// When set, this flag indicates that the first physical sector of the device
+// is aligned with the first logical sector. When not set, the first physical
+// sector of the device is misaligned with the first logical sector.
+
+#define	SSINFO_FLAGS_PARTITION_ALIGNED_ON_DEVICE	0x00000002
+// When set, this flag indicates that the partition is aligned to physical
+// sector boundaries on the storage device.
+
+#define	SSINFO_FLAGS_NO_SEEK_PENALTY	0x00000004
+// When set, the device reports that it does not incur a seek penalty (this
+// typically indicates that the device does not have rotating media, such as
+// flash-based disks).
+
+#define	SSINFO_FLAGS_TRIM_ENABLED	0x00000008
+// When set, the device supports TRIM operations, either T13 (ATA) TRIM or
+// T10 (SCSI/SAS) UNMAP.
+
+#define	SSINFO_OFFSET_UNKNOWN		0xffffffff
+// For "Alignment" fields below
+
+/* We have to lie to Windows Hyper-V about our logical record size. */
```

Comment:
Did I miss a comment somewhere about why we have to lie?

Reply:
... because with larger sizes it fails setting up a virtual disk.